src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -121,6 +121,8 @@ private BoundExpression MakeIsPatternExpression(
                 }
             }
 
+            DecisionDagBuilder.CheckRedundantPatternsForIsPattern(this.Compilation, pattern.Syntax, expression, pattern, diagnostics);
```

Comment:
>DecisionDagBuilder.CheckRedundantPatternsForIsPattern(this.Compilation, pattern.Syntax, expression, pattern, diagnostics); [](http://example.com/codeflow?start=12&length=122)

Perhaps we should skip this check if `diagnostics` doesn't actually accumulate diagnostics.